### PR TITLE
fix softmax cross entropy integer overflow

### PR DESCRIPTION
**Pull request by ZHUI**
### PR types

Bug fixes
### PR changes

OPs
### Describe

fix the integer overflow bug of softmax cross entropy op.

### CODE REVIEW

File: paddle/fluid/operators/softmax_with_cross_entropy_op.cu [hunk 1]
```diff
@@ -22,27 +22,27 @@ using Tensor = framework::Tensor;
...
+                                 const size_t remain, const int ignore_index) {
+  CUDA_KERNEL_LOOP_TYPE(index, n * remain, int64_t) {
+    size_t idx_n = index / remain;
+    size_t idx_remain = index % remain;
+    size_t tmp = labels[index];
     if (ignore_index != tmp) {
-      int idx = idx_n * d + tmp * remain + idx_remain;
+      size_t idx = idx_n * d + tmp * remain + idx_remain;
       logit_grad[idx] -= static_cast<T>(1.);
     }
```
Comment (wawltor) [inline paddle/fluid/operators/softmax_with_cross_entropy_op.cu:34]: 这里的写法有点混乱，要么size_t，要么int64

**Review by chenwhql — APPROVED** [review]
LGTM

**Comment by ZHUI** [comment]
@GaoWei8 This PR simply changed some data types from `int` to `int64_t`.  Since the maximum value of `int`  is 2^31-1,  if there are more than  2^31-1 elements in  the input data, the original code will cause illegal memory access error. 
This change of data type seems to affect the performance of the `softmax_with_cross_entropy_1` test case in PR-CI-OP-benchmark. 
However, the overflow problem should be resolved to support larger input data. We hope to skip the PR-CI-OP-benchmark sincerely.

**Review by GaoWei8 — APPROVED** [review]
LGTM for op benchmark

**Review by wawltor — APPROVED** [review]
LGTM
